src/osd/PG.cc

```diff
@@ -4322,6 +4349,12 @@ void PG::scrub_compare_maps()
 
   // ok, do the pg-type specific scrubbing
   _scrub(authmap, missing_digest);
+  if (!scrubber.store->empty()) {
+    dout(10) << __func__ << ": updating scrub object" << dendl;
+    ObjectStore::Transaction t;
+    scrubber.store->flush(&t);
+    osd->store->apply_transaction(osr.get(), std::move(t));
```

Comment:
We still can't wait here, you must call queue_transaction rather than apply_transaction.  Generally, it's not a good idea to call apply_transaction while sitting on the pg lock.